### ISSUE TITLE
finding only one source with catalogue-based source finder

### ISSUE THREAD

**Issue by panstarrs** [SoFiA-Admin/SoFiA-2#82]
Hello, I recently use catalogue-based source finding https://github.com/SoFiA-Admin/OptiFind to compute HI fluxes, but it looks OptiFind may find multiple sources even within a small volume depending on the preset parameters. My question is if we can just let SoFiA-2 only find one source in a pre-defined region by parameters <r_spat> and <r_spec> and RA, Dec, freq, of course. In this case, the total flux could be simply the summation of all pixels within that region. This function is naive as no source-finding process is actually required, but it could be useful for making catalogs including very faint HI sources (e.g. found only by eyeballing), and for stacking studies below the detection threshold where an optical catalogue is available already. I think there are quite a few existing functions of SoFiA-2 for removing RFI and artefacts can also be useful in this case. I'm not sure if this function of finding only one source can be accomplished by extending the merging areas, i.e. reducing the merging link to zero ? thanks

**Comment by SoFiA-Admin** [comment]
Thank you for this suggestion. I am not sure if such a feature would be particularly useful from a source finding perspective. If you are simply interested in extracting the total flux within a box around a given position and frequency, then this could be easily achieved with a simple three-liner in Python:
```python
from astropy.io import fits
data = fits.open("data.fits")[0].data
data[20:30,20:30,20:30].sum()
```
The WCS module in Astropy will allow you to work out the region boundaries for any position in celestial coordinates.

In principal, you can also achieve the same with the current SoFiA version by setting your OptiFind regions as desired and using SoFiA’s threshold finder with a very small absolute flux threshold (e.g., `1e-30`) that is guaranteed to pick up all pixels within that region. In addition, you may want to set `linker.keepNegative = true` to also retain detections with overall negative flux, if desired.

So, in summary I don’t think that there is a need for such a feature given that this can easily be achieved with the existing functionality in either SoFiA or Python, but let me know if you have any additional requirements that aren’t covered by the above.

**Comment by panstarrs** [comment]
Thanks, and all sounds good to me. I'll let you know if anything comes up later.
